File: server/api/src/main/kotlin/com/company/ipcamera/server/service/ReportService.kt
package com.company.ipcamera.server.service

/**
 * Сервис генерации аналитических отчётов (блок 8.4).
 * Экспорт в CSV, PDF и др. по типам отчётов (события, записи, ANPR, сводка).
 */
interface ReportService {

    enum class ReportType {
        EVENTS_SUMMARY,
        RECORDINGS_SUMMARY,
        LICENSE_PLATES_SUMMARY,
        ANALYTICS_DASHBOARD
    }

    enum class ExportFormat {
        CSV,
        PDF
    }

    /**
     * Сгенерировать отчёт и вернуть содержимое файла.
     * @param type тип отчёта
     * @param format формат экспорта
     * @param cameraId опционально — фильтр по камере
     * @param fromTimestamp начало периода (мс)
     * @param toTimestamp конец периода (мс)
     */
    suspend fun generateReport(
        type: ReportType,
        format: ExportFormat,
        cameraId: String? = null,
        fromTimestamp: Long? = null,
        toTimestamp: Long? = null
    ): Result<ByteArray>

    /** Рекомендуемое расширение файла для формата */
    fun fileExtension(format: ExportFormat): String =
        when (format) {
            ExportFormat.CSV -> "csv"
            ExportFormat.PDF -> "pdf"
        }
}

File: server/web/src/services/reportService.ts
import apiClient from '@/utils/api';
import type { Report, ReportFilter, ExportFormat } from '@/types';

export interface ReportParams {
  type: 'EVENTS_SUMMARY' | 'RECORDINGS_SUMMARY' | 'LICENSE_PLATES_SUMMARY' | 'ANALYTICS_DASHBOARD';
  format: 'CSV' | 'PDF';
  cameraId?: string;
  fromTimestamp?: number;
  toTimestamp?: number;
}

const reportService = {
  /**
   * Получить список отчётов
   */
  async getReports(filter: ReportFilter): Promise<Report[]> {
    const response = await apiClient.get<Report[]>('/reports', { params: filter });
    if (response.data) {
      return Array.isArray(response.data) ? response.data : [];
    }
    return [];
  },

  /**
   * Получить отчёт по ID
   */
  async getReportById(id: string): Promise<Report> {
    const response = await apiClient.get<Report>(`/reports/${id}`);
    return response.data;
  },

  /**
   * Сгенерировать новый отчёт
   */
  async generateReport(data: Partial<Report>): Promise<Report> {
    const response = await apiClient.post<Report>('/reports', data);
    return response.data;
  },

  /**
   * Экспортировать отчёт
   */
  async exportReport(format: ExportFormat): Promise<void> {
    const response = await apiClient.get(`/reports/export?format=${format}`, {
      responseType: 'blob',
    });
    
    // Создаём ссылку для скачивания
    const blob = new Blob([response.data], { type: `application/${format}` });
    const url = window.URL.createObjectURL(blob);
    const link = document.createElement('a');
    link.href = url;
    link.download = `report.${format}`;
    document.body.appendChild(link);
    link.click();
    document.body.removeChild(link);
    window.URL.revokeObjectURL(url);
  },
};

export { reportService };
